sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/AttributeAssertion.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.testing.assertj;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.common.AttributeKey;
+import java.util.function.Consumer;
+import org.assertj.core.api.AbstractAssert;
+
+/** An assertion on an attribute key. */
+@AutoValue
+public abstract class AttributeAssertion {
+
+  // This method is not type-safe! But because the constructor is private, we know it will only be
```

Comment:
The main purpose of having this class instead of using Map.Entry is to be able to make the constructor private